pkg/rehearse/jobs.go

```diff
@@ -396,6 +396,64 @@ func AddRandomJobsForChangedTemplates(templates []config.ConfigMapSource, toBeRe
 	return rehearsals
 }
 
+func getPresubmitForRegistryStep(step config.RegistryStep, configs config.ByFilename, prConfigPresubmits map[string][]prowconfig.Presubmit) (string, *prowconfig.Presubmit) {
+	for _, ciopConfig := range configs {
+		tests := ciopConfig.Configuration.Tests
+		repoPresubmits := prConfigPresubmits[ciopConfig.Info.Repo]
+		for _, test := range tests {
+			if test.MultiStageTestConfiguration != nil {
+				jobPrefix := fmt.Sprintf("%s-ci-%s-%s-%s-", "pull", ciopConfig.Info.Org, ciopConfig.Info.Repo, ciopConfig.Info.Branch)
+				jobName := fmt.Sprintf("%s%s", jobPrefix, test.As)
+				if step.Workflow != nil && test.MultiStageTestConfiguration.Workflow != nil && step.Workflow.As == *test.MultiStageTestConfiguration.Workflow {
+					for _, presubmit := range repoPresubmits {
+						if presubmit.Name == jobName {
+							return ciopConfig.Info.Repo, &presubmit
+						}
+					}
+					// continue to check other tests
+					continue
```

Comment:
The control flow here is a bit odd. I suggest using a `switch` to make the algorithm more clear -- you have already pruned the diff so each change is unique, so every job you look at can match only one of the changed items (workflow,chain,ref)